iml-agent/src/daemon_plugins/corosync.rs

```diff
@@ -28,7 +28,14 @@ impl DaemonPlugin for Corosync {
         &self,
     ) -> Pin<Box<dyn Future<Output = Result<Output, ImlAgentError>> + Send>> {
         async move {
-            let x = get_crm_mon().await?;
+            let node_id = get_local_nodeid().await?;
+
+            let cluster = get_crm_mon().await?;
+
+            let x = match (node_id, cluster) {
+                (Some(x), Some(y)) => Some((x, y)),
```

Comment:
I believe this feature is still in nightly but eventually, we should be able to do something like:
```
let x: Option<_> = try { (node_id?, cluster?) };
```

Reply:
There is `option.zip` that landed in 1.46 and I'd like to use. However we need to wait till epel bumps the rust compiler.